### PR TITLE
Requires check

### PR DESCRIPTION
**Pull request by irushchyshyn**
- [x] check if the package uses versioned Python prefix in requirements' names
- [x] add integration tests for this check
- [x] modify the shared `is_unversioned` function to disregard requirements like `/usr/libexec/system-python` or
`libsamba-python-samba4.so` (it should be ok to fix this directly in the shared function as it should be propagated to portingdb as well)
- [x] Check also Build requires
- [x] Unit tests

~~Note: this task will fail even if the correct name is not provided by the required package. And I am not sure if we can check if it is provided. Maybe we could change the outcome to smth like NEEDS_INSPECTION, or look for a way to check this. Any thoughts?~~

As part of #5.

### CODE REVIEW

File: taskotron_python_versions/requires.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+MESSAGE = """These RPMs use `python-` prefix without Python version in Requires:
+{}
+This is strongly discouraged and should be avoided. Please check
+the required packages, and use names with either `python2-` or
+`python3-` prefix if available.
+"""
+
+# Should be a link to guidelines when
+# https://pagure.io/packaging-committee/issue/686 accepted.
+INFO_URL = ''
```
Comment (irushchyshyn) [inline taskotron_python_versions/requires.py]: This should probably be a link to guidelines, when https://pagure.io/packaging-committee/issue/686 is considered.

File: taskotron_python_versions/requires.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+
+def task_requires(packages, koji_build, artifact):
```
Comment (hroncok) [inline taskotron_python_versions/requires.py]: task_requires could mean anything. What about `task_requires_naming_scheme`? Applies to all other names in this PR.

**Comment by hroncok** [comment]
What about the following?

 1. Given the python name, try to fix it to a python2 name.
     * (If not possible for given name, return NEEDS_INSPECTION)
 2. Run dnf repoquery to check if such package is available.

**Comment by irushchyshyn** [comment]
>    1. Given the python name, try to fix it to a python2 name.
>         (If not possible for given name, return NEEDS_INSPECTION)
>     2. Run dnf repoquery to check if such package is available.

That might work, but what if:
1) Given the requirement name, run a dnf repoquery on what provides it.  It will return the name of the package, and we can check if it is not misnamed. For example:
```
$ dnf repoquery --whatprovides rpm-python
python2-rpm-0:4.13.0-0.rc1.46.fc25.x86_64
python2-rpm-0:4.13.0.1-1.fc25.x86_64
```
This way we do not have to guess what the python2 name would look like. (step 1). @hroncok how about that?

Also, will it be enough to run dnf repoquery with `--releasever` so that it queries the exact Fedora release it is built for?

File: taskotron_python_versions/requires.py (hunk 2 above)
Comment (irushchyshyn) [inline taskotron_python_versions/requires.py]: Renamed in  e7a1245

**Comment by hroncok** [comment]
> This way we do not have to guess what the python2 name would look like. (step 1). @hroncok how about that?

Cool!

> Also, will it be enough to run dnf repoquery with --releasever so that it queries the exact Fedora release it is built for?

I think so.

**Comment by hroncok** [comment]
@irushchyshyn  Rebased. Could you have a look?

We also need a link there, so let's just use the ticket URL for now maybe?

**Comment by hroncok** [comment]
Repoquery from Python:

```python
import platform
import sys

import dnf

base = dnf.Base()
base.conf.substitutions['releasever'] = platform.linux_distribution()[1]
base.read_all_repos()
base.fill_sack(load_system_repo=True, load_available_repos=True)
q = base.sack.query()
q = q.filter(provides__glob=[sys.argv[1]])
packages = q.run()

for package in packages:
    print('{p.name} from {p.reponame} repo'.format(p=package))
```

**Comment by irushchyshyn** [comment]
@hroncok thanks for rebasing. Looks good.

> Repoquery from Python:
> ...

Thanks for this. I have been working on something similar, although can we rely on `platform.linux_distribution()[1]`? I am extracting release version from package release for now.

I'll clean up my change  and will push it hopefully soon.

**Comment by hroncok** [comment]
> although can we rely on `platform.linux_distribution()[1]`? I am extracting release version from package release for now.

No, we cannot. This was just an example. Extracting the release version from the package release is a right call.

File: taskotron_python_versions/requires.py [hunk 3]
```diff
@@ -23,24 +93,24 @@ def task_requires_naming_scheme(packages, koji_build, artifact):
     # to make the above functions testable anyway
     from libtaskotron import check
 
+    _, fedora_release = koji_build.split('fc')
```
Comment (hroncok) [inline taskotron_python_versions/requires.py]: Not good.

```python
koji_build = 'xfce4-terminal-0.8.5.1-1.fc26.x86_64'
_, fedora_release = koji_build.split('fc')
```

Comment (hroncok) [inline taskotron_python_versions/requires.py]: I'd go with:

```python
koji_build.split('fc')[-1]
```

Comment (hroncok) [inline taskotron_python_versions/requires.py]: I'll commit that.

**Comment by hroncok** [comment]
Oh. So the problem with python2-lxml is it's only in updates-testing. That's why the test work on my machine but fail in Travis.

File: test/functional/test_requires.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+from taskotron_python_versions.requires import (
+    get_versioned_name,
+    check_requires_naming_scheme,
+)
+
+from .common import gpkg
+
+
+Package = namedtuple('Pcakage', 'name')
```
Comment (hroncok) [inline test/functional/test_requires.py]: typo Pcakage

File: taskotron_python_versions/requires.py [hunk 5]
```diff
@@ -82,7 +82,6 @@ def check_requires_naming_scheme(package, repoquery):
                     'available'.format(package.filename, name, versioned))
                 misnamed_requires.add('{} ({} is available)'.format(
                     name, versioned))
-
```
Comment (hroncok) [inline taskotron_python_versions/requires.py]: Could you please rebase those unrelated changes to their appropriate commits?

File: test/functional/test_requires.py [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+        self.data = data
+        self.provides = None
+
+    def filter(self, provides):
+        self.provides = provides
+        return self
+
+    def run(self):
+        if self.provides not in self.data:
+            raise Exception('Running repoquery on {} is not expected'.format(
```
Comment (hroncok) [inline test/functional/test_requires.py]: Is the general Exception here on purpose or just for the lack of a more specific one? I'd go with ValueError here I guess.

File: test/functional/test_requires.py [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+        'python-decoratortools': ['python-decoratortools'],
+        'python-peak-util-addons': ['python-peak-util-addons'],
+        'python-peak-util-assembler': ['python-peak-util-assembler'],
+        'python-peak-util-extremes': ['python-peak-util-extremes'],
+    })),
+))
+def test_package_requires_are_correct(pkgglob, repoquery,):
+    assert not check_requires_naming_scheme(gpkg(pkgglob), repoquery)
+    assert not repoquery.data, (
+        'Repoquery was not called for: {}'.format(repoquery.data))
```
Comment (hroncok) [inline test/functional/test_requires.py]: This asserts all items from `repoquery.data` were popped out? If so, could you please add a comment about that? Because I don't find it very obvious.

File: test/functional/test_requires.py (hunk 6 above)
Comment (irushchyshyn) [inline test/functional/test_requires.py]: For the lack of a more specific one. `ValueError` should go as well, will change

File: test/functional/test_requires.py (hunk 7 above)
Comment (irushchyshyn) [inline test/functional/test_requires.py]: Yes. Added a comment and a docstring for the run method.

File: taskotron_python_versions/requires.py (hunk 5 above)
Comment (irushchyshyn) [inline taskotron_python_versions/requires.py]: Sure, rebased

**Comment by hroncok** [comment]
Excellent! Let's watch http://taskotron-dev.fedoraproject.org/resultsdb/results?testcases=dist.python-versions and see how it actually works.

**Comment by irushchyshyn** [comment]
@hroncok Thanks a lot for review and contributions!
